docs/usage/CART_DISCOUNT_SYNC.md

```diff
@@ -67,28 +67,40 @@ Therefore, in order to resolve the actual ids of those references in the sync pr
+When syncing from a source commercetools project, you can use [`transformCartDiscountReferences`](https://commercetools.github.io/commercetools-sync-java/v/4.0.1/com/commercetools/sync/cartdiscounts/service/CartDiscountReferenceTransformService.html#transformCartDiscountReferences-java.util.List-)
+method that transforms(resolves by querying and caching key-id pairs) and maps from a `CartDiscount` to `CartDiscountDraft` using cache in order to make them ready for reference resolution by the sync, for example: 
 
 ````java
-// Build a CartDiscountQuery for fetching cart discounts from a source CTP project with all the needed references expanded for the sync
-final CartDiscountQuery cartDiscountQueryWithReferenceExpanded = CartDiscountReferenceResolutionUtils.buildCartDiscountQuery();
+// Build a CartDiscountQuery for fetching cart discounts from a source CTP project without any references expanded for the sync:
+final CartDiscountQuery cartDiscountQueryWithReferences = CartDiscountReferenceResolutionUtils.buildCartDiscountQuery();
 
 // Query all cart discounts (NOTE this is just for example, please adjust your logic)
 final List<CartDiscount> cartDiscounts =
     CtpQueryUtils
-        .queryAll(sphereClient, cartDiscountQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, cartDiscountQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the cartDiscount, 
+Initialize [`CartDiscountReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/cartdiscounts/service/CartDiscountReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU (Last Recently Used) based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the `transformCartDiscountReferences` method with the `cartDiscounts` parameter as shown below:
 
-// Mapping from CartDiscount to CartDiscountDraft with considering reference resolution.
-final List<CartDiscountDraft> cartDiscountDrafts = CartDiscountReferenceResolutionUtils.mapToCartDiscountDrafts(cartDiscounts);
+````java
+// Fetch(Id to key values for references) into the cache and map from CartDiscount to CartDiscountDraft using cache with considering reference resolution.
+final List<CartDiscountDraft> cartDiscountDrafts = CartDiscountReferenceTransformService.transformCartDiscountReferences(cartDiscounts);
```

Comment:
`transformCartDiscountReferences` does not return a CompletableFuture? Also, naming is a bit confusing as in the end the service transforms a cart discount to cart discount draft and reference resolution is just part of the process. Just an idea:

`CartDiscount[Service/Utils].toCartDiscountDrafts(cartDiscounts);`

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/710/commits/877461e014cc78caa3af31ea2fb883ff1f5aa05e

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/710/commits/b7c447212e690ccc9422b14dc2555f6553bf602f